vnext/Desktop.ABITests/DynamicReaderWriterTests.cpp

```diff
@@ -0,0 +1,178 @@
+
+    TestCheck(reader.GetNextArrayItem());
+    TestCheckEqual(JSValueType::Boolean, reader.ValueType());
+    TestCheckEqual(true, reader.GetBoolean());
+
+    TestCheck(reader.GetNextArrayItem());
+    TestCheckEqual(JSValueType::Int64, reader.ValueType());
+    TestCheckEqual(123, reader.GetInt64());
+
+    TestCheck(reader.GetNextArrayItem());
+    TestCheckEqual(JSValueType::String, reader.ValueType());
+    TestCheckEqual(L"abc", reader.GetString());
+
+    TestCheck(!reader.GetNextArrayItem());
+  }
+
+  TEST_METHOD(WriteGetObject) {
+    IJSValueWriter writer = Microsoft::Internal::TestController::CreateDynamicWriter();
+
+    auto namePropertyName = to_hstring(L"Name");
+    auto agePropertyName = to_hstring(L"Age");
+    auto marriedPropertyName = to_hstring(L"Married");
+    auto emptyPropertyName = to_hstring(L"");
+
+    writer.WriteObjectBegin();
+
+    writer.WritePropertyName(namePropertyName);
+    writer.WriteString(L"Bob");
+
+    writer.WritePropertyName(agePropertyName);
+    writer.WriteInt64(32);
+
+    writer.WritePropertyName(marriedPropertyName);
+    writer.WriteBoolean(true);
+
+    writer.WriteObjectEnd();
+
+    IJSValueReader reader = Microsoft::Internal::TestController::CreateDynamicReader(writer);
+    TestCheckEqual(JSValueType::Object, reader.ValueType());
+
+    TestCheck(reader.GetNextObjectProperty(namePropertyName));
+    TestCheckEqual(JSValueType::String, reader.ValueType());
+    TestCheckEqual(L"Bob", reader.GetString());
+
+    TestCheck(reader.GetNextObjectProperty(agePropertyName));
+    TestCheckEqual(JSValueType::Int64, reader.ValueType());
+    TestCheckEqual(32, reader.GetInt64());
+
+    TestCheck(reader.GetNextObjectProperty(marriedPropertyName));
+    TestCheckEqual(JSValueType::Boolean, reader.ValueType());
+    TestCheckEqual(true, reader.GetBoolean());
+
+    TestCheck(!reader.GetNextObjectProperty(emptyPropertyName));
+  }
+
+  TEST_METHOD(WriteGetObjectArrayMix) {
+    IJSValueWriter writer = Microsoft::Internal::TestController::CreateDynamicWriter();
+
+    auto namePropertyName = to_hstring(L"Name");
+    auto countiesPropertyName = to_hstring(L"Counties");
+    auto areaPropertyName = to_hstring(L"Area");
+    auto emptyPropertyName = to_hstring(L"");
+
+    writer.WriteObjectBegin();
+
+    writer.WritePropertyName(namePropertyName);
+    writer.WriteString(L"Washington");
+
+    writer.WritePropertyName(countiesPropertyName);
+    writer.WriteArrayBegin();
+    writer.WriteString(L"Snohomish");
+    writer.WriteString(L"King");
+    writer.WriteString(L"Pierce");
+    writer.WriteArrayEnd();
+
+    writer.WritePropertyName(areaPropertyName);
+    writer.WriteInt64(184827);
+
+    writer.WriteObjectEnd();
+
+    IJSValueReader reader = Microsoft::Internal::TestController::CreateDynamicReader(writer);
+    TestCheckEqual(JSValueType::Object, reader.ValueType());
+
+    TestCheck(reader.GetNextObjectProperty(namePropertyName));
```

Comment:
>namePropertyName [](start = 43, length = 16)

my understanding is that this is an out value to the properName.
therefore we should probablye declare an `hstring currentProperty` and pass that here and have a `TestCheckEqual(namePropertyName, currentProperty)`

Reply:
That would make sense, let me try.